### PR TITLE
Add more component tests for controls

### PR DESCRIPTION
**Pull request by kienstra**
#### Changes
* Adds component tests for all controls that didn't have them
* Continues #514
* Should give us even more confidence in each commit. Also, we should be able to merge a lot of `dependabot` PRs without any manual testing.

#### Testing instructions
1. `npm install`
1.  `npm run test:js`
1. Expected: there are only dots, no error messages
1. Verify that `classic-text`, `range`, and `select` controls still work (they have minor edits here)

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -4,13 +4,17 @@
     "plugin:@wordpress/eslint-plugin/recommended",
     "plugin:import/recommended",
     "plugin:eslint-comments/recommended",
-    "plugin:jsx-a11y/recommended"
+    "plugin:jsx-a11y/recommended",
+    "plugin:testing-library/recommended",
+    "plugin:jest-dom/recommended"
```
Comment (kienstra) [inline .eslintrc:9]: Linting plugins for the JS tests

File: bin/pre-commit.sh [hunk 2]
```diff
@@ -13,6 +13,7 @@ fi
 js_files=$( git diff --diff-filter=d --staged --name-only | grep -E '^js\/\S*\.js$' )
 if [ ! -z "$js_files" ]; then
     npm run lint:js:files $js_files
+    npm run test:js $js_files -- --findRelatedTests
```
Comment (kienstra) [inline bin/pre-commit.sh:16]: In the pre-commit hook, this runs all Jest tests related to the staged files. So if only the `text.js` file is staged, it'll only run the test for that.

File: js/blocks/controls/classic-text.js [hunk 3]
```diff
@@ -9,16 +9,15 @@ import { BaseControl } from '@wordpress/components';
...
-	const { clientId } = parentBlockProps;
+	const { field, getValue, onChange, rowIndex, parentBlockProps: { clientId } } = props;
 	const editorId = 'number' === typeof rowIndex ? `bl-${ clientId }-${ field.name }-rowIndex-${ rowIndex }` : `bl-${ clientId }-${ field.name }`;
 	const initialValue = getValue( props );
 	const value = 'undefined' !== typeof initialValue ? initialValue : field.default;
 
 	return (
 		<BaseControl
 			label={ field.label }
-			id={ `bl-classic-text-${ instanceId }` }
```
Comment (kienstra) [inline js/blocks/controls/classic-text.js:21]: `instanceId` was undefined

File: js/blocks/controls/range.js [hunk 4]
```diff
@@ -9,16 +9,13 @@ const BlockLabRangeControl = ( props ) => {
 
 	return (
 		<RangeControl
-			beforeIcon={ field.beforeIcon }
-			afterIcon={ field.afterIcon }
```
Comment (kienstra) [inline js/blocks/controls/range.js:13]: These properties of `field` don't exist

File: js/blocks/controls/select.js [hunk 5]
```diff
@@ -2,14 +2,13 @@
...
 import { SelectControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
 
 const BlockLabSelectControl = ( props ) => {
 	const { field, getValue, onChange } = props;
-	const { __ } = wp.i18n;
 
 	if ( '' === field.default ) {
 		field.options = [
-			// @see https://github.com/WordPress/gutenberg/issues/11270 Disabled attribute not currently supported.
```
Comment (kienstra) [inline js/blocks/controls/select.js:12]: This issue looks to be resolved

File: tests/js/jest.setup.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+/**
+ * External dependencies
+ */
+import '@testing-library/jest-dom/extend-expect';
```
Comment (kienstra) [inline tests/js/jest.setup.js:4]: This prevents having to import this package in every test file

**Comment by kienstra** [comment]
## Request For Review

Hi @lukecarbis,
Hope you're doing well.

Could you please review this when you have a chance? The diff is almost entirely test files, but there are minor edits in the `classic-text`, `range,` and `select` controls.

This should give us even more confidence in each commit. Every control now has a component test.

Thanks, Luke!

File: js/blocks/controls/range.js (hunk 4 above)
Comment (lukecarbis) [reply js/blocks/controls/range.js:13]: Keeping up with Gutenberg development is hard!

**Comment by lukecarbis** [comment]
Amazing work!

**Review by lukecarbis — APPROVED** [review]

**Comment by kienstra** [comment]
Thanks, Luke!

File: js/blocks/controls/range.js (hunk 4 above)
Comment (kienstra) [reply js/blocks/controls/range.js:13]: Haha, yeah. I don't think this caused any problem 😄

**Pull request merged**